opm/autodiff/FlowMain.hpp

```diff
@@ -391,6 +391,8 @@ namespace Opm
                 deck_ = parser->parseFile(deck_filename, parseContext);
                 checkDeck(deck_, parser);
                 eclipse_state_.reset(new EclipseState(deck_, parseContext));
+                auto ioConfig = eclipse_state_->getIOConfig();
+                ioConfig->setOutputDir(output_dir_);
```

Comment:
>>  There are some things about const and pointers that are not so intuitive,
>
> I don't understand it ...

constness can even get downright deceptive. consider the following example:

```c++
#include <iostream>

struct A
{ 
 A(const int& v) : v_(v) {};
 void print() const { std::cout << v_ << "\n"; }
 const int& v_;
};

int main()
{
  int foo = 123;
  A a(foo);
  foo = 456;
  a.print();
  return 0;
}
```

one might expect that this prints `123` but it doesn't: it prints `456`. (As a side note, some old version of dune-pdelab took advantage of this. As you can imagine it took me days of debugging to find it out...) But it gets even worse: if you use

```c++
A a(123);
```

things will still compile, but you get undefined behavior. (which is most likely the one which you expect, but just by coincidence.)